samples/bluetooth/broadcast_audio_sink/src/main.c

```diff
@@ -83,99 +122,250 @@ static uint8_t sink_broadcast_code[BT_AUDIO_BROADCAST_CODE_SIZE];
 
 uint64_t total_rx_iso_packet_count; /* This value is exposed to test code */
 
-#if defined(CONFIG_LIBLC3)
+#if defined(CONFIG_USB_DEVICE_AUDIO)
+static int16_t usb_audio_data[MAX_NUM_SAMPLES_STEREO] = {0};
+static int16_t usb_audio_data_stereo[MAX_NUM_SAMPLES_STEREO] = {0};
 
-#include "lc3.h"
+RING_BUF_DECLARE(ring_buf_usb, AUDIO_RING_BUF_SIZE);
+NET_BUF_POOL_DEFINE(usb_tx_buf_pool, USB_TX_BUF_NUM, BROADCAST_STEREO_SAMPLE_SIZE, 0,
+		    net_buf_destroy);
 
-#define MAX_SAMPLE_RATE       16000
-#define MAX_FRAME_DURATION_US 10000
-#define MAX_NUM_SAMPLES       ((MAX_FRAME_DURATION_US * MAX_SAMPLE_RATE) / USEC_PER_SEC)
+static void mix_mono_to_stereo(enum bt_audio_location channels);
+#endif /* defined(CONFIG_USB_DEVICE_AUDIO) */
 
-static int16_t audio_buf[MAX_NUM_SAMPLES];
-static lc3_decoder_t lc3_decoder;
-static lc3_decoder_mem_16k_t lc3_decoder_mem;
+#if defined(CONFIG_LIBLC3)
+static int16_t audio_buf[MAX_NUM_SAMPLES_MONO];
 static int frames_per_sdu;
+static K_SEM_DEFINE(lc3_decoder_sem, 0, 1);
+
+static void do_lc3_decode(struct broadcast_sink_stream *sink_stream);
+static void lc3_decoder_thread(void *arg1, void *arg2, void *arg3);
+K_THREAD_DEFINE(decoder_tid, LC3_ENCODER_STACK_SIZE, lc3_decoder_thread,
+		NULL, NULL, NULL, LC3_ENCODER_PRIORITY, 0, -1);
+
+/* Consumer thread of the decoded stream data */
+static void lc3_decoder_thread(void *arg1, void *arg2, void *arg3)
+{
+	while (true) {
+		k_sem_take(&lc3_decoder_sem, K_FOREVER);
+		/* For now we only handle one BIS, so always only decode the first element in
+		 * streams.
+		 */
+		do_lc3_decode(&streams[0]);
+#if defined(CONFIG_USB_DEVICE_AUDIO)
+		int err = 0;
+		enum bt_audio_location channels;
+
+		err = bt_audio_codec_cfg_get_chan_allocation(streams[0].stream.codec_cfg,
+							     &channels);
+
+		/* If the ring buffer usage is alrger than zero, then there is data to process */
+		if (ring_buf_space_get(&streams[0].audio_ring_buf)) {
+			mix_mono_to_stereo(channels);
+		}
+#endif /* #if defined(CONFIG_USB_DEVICE_AUDIO) */
```

Comment:
While I understand that decoding multiple streams for USB is an issue, this also effectively modify the non-USB configuration to only support a single stream, compared to the current version of the sample that supports decoding multiple stream. 

I do not think the non-USB configuration should reduce its capabilities.

Reply:
I'm not entirely sure what you are referring to here. The code snippet you are highlighting here only runs for USB. What runs above is what ran prior to this PR. 
Assuming the Kconfigs you mentioned above are dependent on USB also, then there can also not be any modications to hos it syncs to the stream.

Reply:
Sorry, it may not have been very clear:

This in a generic function that runs for both USB and non-USB builds, and you ever only do `do_lc3_decode(&streams[0]);` so this effectively reduces the sample's capabilities to decode all streams to just the first stream

Reply:
Aah, yes, thats correct, I will fix this